### PR TITLE
Top overlay in embedded player

### LINKED ISSUE #2935: Top of the embedded player overlay
## Scope
Implement Embedded player top overlay with sharing button 

## AC
- [ ] Embedded version has top of screen overlay
- [ ] Sharing button click summons sharing screen

Out of scope:

- #2936 
- https://github.com/Joystream/atlas/issues/2943

## Design
Screenshot 2022-06-29 at 17 27 28

[Figma link >](https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=2583%3A232036)

### PR DESCRIPTION
**Pull request by drillprop**
Fix #2935

### CODE REVIEW

File: packages/atlas/src/components/_video/VideoPlayer/PlayerControlButton.styles.ts [hunk 1]
```diff
@@ -119,6 +119,14 @@ export const ControlButtonTooltip = styled.div<ControlButtonTooltipProps>`
   padding: 0.5em;
   white-space: nowrap;
   transition: opacity ${transitions.timings.player} ease-in, backdrop-filter ${transitions.timings.player} ease-in;
+
+  ${({ tooltipPosition }) =>
```
Comment (rafalpawlow) [inline packages/atlas/src/components/_video/VideoPlayer/PlayerControlButton.styles.ts]: Can we make some `getTooltipPosition` function which will cover all scenarios?

File: packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts [hunk 2]
```diff
@@ -91,9 +93,51 @@ export const StyledSvgControlsSettingsSolid = styled(SvgControlsSettingsSolid)`
...
   ${defaultIconColor};
 `
+export const StyledSvgControlsShare = styled(SvgControlsShare)`
+  ${defaultIconColor};
+`
 
 export const TRANSITION_DELAY = '50ms'
 
+export const EmbbeddedTopBar = styled.div<{ isFullScreen: boolean }>`
+  font-size: ${sizes(4)};
```
Comment (rafalpawlow) [inline packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts]: Didn't we agree, we use `sizes` only for margins and paddings? Ignore this if I'm wrong

File: packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts [hunk 3]
```diff
@@ -91,9 +93,51 @@ export const StyledSvgControlsSettingsSolid = styled(SvgControlsSettingsSolid)`
...
+  display: grid;
+  grid-template-columns: auto 1fr auto;
+  align-items: center;
+  padding: 0.5em 0.5em 0.5em 1em;
+  gap: 1em;
+  opacity: 0;
+  visibility: hidden;
+  transition: opacity, visibility;
+  transition-delay: ${TRANSITION_DELAY};
+  transition-duration: 200ms;
```
Comment (rafalpawlow) [inline packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts]: Couldn't we use some design token here with similar values?

File: packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts [hunk 4]
```diff
@@ -91,9 +93,51 @@ export const StyledSvgControlsSettingsSolid = styled(SvgControlsSettingsSolid)`
...
+  transition-timing-function: ${transitions.easing};
+  @media (hover: hover) {
+    font-size: ${({ isFullScreen }) => (isFullScreen ? sizes(8) : sizes(4))};
```
Comment (rafalpawlow) [inline packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts]: Same question about `sizes`

**Review by rafalpawlow — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts (hunk 3 above)
Comment (drillprop) [reply packages/atlas/src/components/_video/VideoPlayer/VideoPlayer.styles.ts]: These values are quite old. Some of them were passed to me during calls, they were never documented if I'm not mistaken. If I change that we would need to go through all the animations in the player and make sure that everything works as expected. We'd need to update also easings and delays. I don't want to bother designers about this right now. 
For now, let's just keep the same values as we have in the controls overlay until we get proper documentation for the player, wdyt?

**Review by rafalpawlow — CHANGES_REQUESTED** [review]
You missed this margin

Zrzut ekranu 2022-07-7 o 10 54 27

**Review by rafalpawlow — APPROVED** [review]

**Pull request merged**